src/observers/qos_correction.cpp

```diff
@@ -2,72 +2,126 @@
 #include <vector>
 #include <utility>
 
-#include "observers/qos_correction.hpp"
-
-#include "mesos/slave/oversubscription.hpp"
+#include "bus/event_bus.hpp"
 
-#include "messages/serenity.hpp"
+#include "observers/qos_correction.hpp"
 
-#include "serenity/wid.hpp"
+#include "serenity/resource_helper.hpp"
 
 namespace mesos {
 namespace serenity {
 
 QoSCorrectionObserver::~QoSCorrectionObserver() {}
 
+Try<Nothing> QoSCorrectionObserver::qosCorrectionStateMachine() {
```

Comment:
```run``` is a good name.
The states are like 'empty contentions', 'cooldown timer', 'new contentions' ... etc.